### PR TITLE
Task-40167: Fix display keyword When the detected keyword on the document contains a capital letter

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

**Review by rdenarie — CHANGES_REQUESTED** [review]
Does your modification works with accentued char ?
Example : in document : "accentué", and in keyword : "accentue"

**Comment by AzmiTouil** [comment]
> Does your modification works with accentued char ?
> Example : in document : "accentué", and in keyword : "accentue"

no , we need to remove accents

File: core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java [hunk 1]
```diff
@@ -208,12 +210,20 @@ private String getDetectedKeywords(Collection<SearchResult> searchResults, Strin
...
                      }
                    }
                  });
     return detectedKeywords.stream().collect(Collectors.joining(", "));
   }
+
+  private String removeAccents(String string) {
+    if (StringUtils.isNotBlank(string)) {
+      string = Normalizer.normalize(string, Normalizer.Form.NFD);
+      string = string.replaceAll("[\\p{InCombiningDiacriticalMarks}]", "");
```
Comment (rdenarie) [inline core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java:225]: are you sure of this line ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java (hunk 1 above)
Comment (AzmiTouil) [reply core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java:225]: Yes

Comment (rdenarie) [reply core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java:225]: Can you explain where it comes from ?

Comment (AzmiTouil) [reply core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java:225]: The \p{InCombiningDiacriticalMarks} will match all such diacritic codes and we will replace them with their deAccented counterparts followed by their combining diacritics.
we have already used it in other places https://github.com/Meeds-io/social/blob/d72cad054a333238e99f0e638f271fd9cc94555c/component/core/src/main/java/org/exoplatform/social/core/jpa/search/ProfileSearchConnector.java#L443

**Review by rdenarie — APPROVED** [review]

**Pull request merged**